### ISSUE TITLE
Experiment:  use validation score as early-stopping criteria, not training score

### ISSUE THREAD

**Issue by nsfabina** [pgbrodrick/rsCNN#58]
Be better data scientists

**Comment by nsfabina** [comment]
@pgbrodrick :  Well, the default metric for this is 'val_loss', so this should be working. Please keep an eye on this and reopen if it's an issue in the future.

**Comment by pgbrodrick** [comment]
I haven't dug into why yet, but this instance appears to demonstrate this is still a problem:

relevant config parameters:
callback_early_stopping:
  use_callback: true
  min_delta: 0.0001
  patience: 5
callback_reduced_learning_rate:
  use_callback: true
  factor: 0.5
  min_delta: 0.0001
  patience: 3

model training (starting part way through)

Epoch 00015: saving model to ./tch_extrapoloation_initial_test/model.h5
Epoch 16/100
1/1 [==============================] - 0s 152ms/step - loss: 0.6484 - val_loss: 0.7784
Epoch 17/100
1/1 [==============================] - 0s 141ms/step - loss: 0.6354 - val_loss: 0.7409
Epoch 18/100
1/1 [==============================] - 0s 193ms/step - loss: 0.6251 - val_loss: 0.7441
Epoch 19/100
1/1 [==============================] - 0s 199ms/step - loss: 0.6129 - val_loss: 0.7929
Epoch 20/100
1/1 [==============================] - 0s 187ms/step - loss: 0.5966 - val_loss: 0.8955

Epoch 00020: saving model to ./tch_extrapoloation_initial_test/model.h5
Epoch 21/100
1/1 [==============================] - 0s 123ms/step - loss: 0.5792 - val_loss: 1.0506
Epoch 22/100
1/1 [==============================] - 0s 172ms/step - loss: 0.5669 - val_loss: 1.2140
Epoch 23/100
1/1 [==============================] - 0s 205ms/step - loss: 0.5608 - val_loss: 1.2920
Epoch 24/100
1/1 [==============================] - 0s 235ms/step - loss: 0.5515 - val_loss: 1.2330
Epoch 25/100
1/1 [==============================] - 0s 168ms/step - loss: 0.5351 - val_loss: 1.1079

Epoch 00025: saving model to ./tch_extrapoloation_initial_test/model.h5
Epoch 26/100
1/1 [==============================] - 0s 122ms/step - loss: 0.5236 - val_loss: 1.0452
Epoch 27/100
1/1 [==============================] - 0s 193ms/step - loss: 0.5150 - val_loss: 1.1047
Epoch 28/100
1/1 [==============================] - 0s 184ms/step - loss: 0.4961 - val_loss: 1.2338
Epoch 29/100
1/1 [==============================] - 0s 197ms/step - loss: 0.4819 - val_loss: 1.2764
Epoch 30/100
1/1 [==============================] - 0s 192ms/step - loss: 0.4719 - val_loss: 1.1605

**Comment by pgbrodrick** [comment]
Fixed, callbacks needed to reference 'val_loss' not 'loss', the latter was hardcoded in.
